app/models/cycle.rb

```diff
@@ -215,6 +215,11 @@ def status
     end
   end
 
+  def color_as_rgba(alpha)
+    rgb = color.scan(/[a-fA-F0-9]{2}/).map { |color| color.to_i(16) }
```

Comment:
I guess this is a presentation concern. You could put this method in a module and use it passing both the color and alpha desired.